immuneML/reports/encoding_reports/DesignMatrixExporter.py

```diff
@@ -50,8 +61,35 @@ def _generate(self) -> ReportResult:
         return ReportResult(self.name, output_tables=[matrix_result], output_text=[details_result, label_result])
 
     def _export_matrix(self) -> ReportOutput:
+        """Create a file for the design matrix in the desired format."""
+        
         data = self._get_data()
-        file_path = self._save_to_file(data, self.result_path / "design_matrix")
+        file_path = self.result_path / "design_matrix"
+
+        # Check that the requested extension is supported.
+        assert self.format.endswith(("hdf5", "hdf5.zip", "csv", "csv.zip", "npy", "npy.zip")), \
```

Comment:
This is my mistake for not mentioning yesterday: could you move this statement to build_object function before the object is instantiated? build_object() is called before any analysis is run, in the parsing time, so if the user did not specify the extension correctly, the error would be raised much earlier and not halfway through potentially long analysis. Also, there is a utility class called ParameterValidator from the util package which can check this and print a nice message if the format is not one of the specified. The docs are here for reference: https://docs.immuneml.uio.no/developer_docs/how_to_add_new_report.html#implementing-the-report :) sorry for the misunderstanding. 🙏

Reply:
If I also will start to say sorry every time I do a mistake (your actually was not a mistake), our chat would become rather boring :) 

Does ParameterValidator from the util package do this already or I have to check this out?

Reply:
Here is an example of how it can be used: [it has multiple utility functions, but you need to specify expected values]

https://docs.immuneml.uio.no/_modules/source/reports/ml_reports/Coefficients.html#Coefficients :)

Reply:
can you check if it ok now?

Reply:
added one line to build_object(), otherwise it's good :)